### PR TITLE
Introducing read all mails

### PR DESCRIPTION
**Pull request by BBeom06**
### Description
 - UI_Mail receive button
 - UI_RewardScreen popup

### Related Link
- [모두받기](https://www.notion.so/planetarium/bcd9cea8e2844551b87363c3751566c5)
- [[건의] 읽지 않은 메일 전체 한번에 읽음처리](https://app.asana.com/0/1202603135170296/1201062951784508/f)

### Screenshot
image
image

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        [SerializeField]
+        private float fadeDuration;
+
+        [SerializeField]
+        private AnimationCurve fadeGraph;
+
+        private int _count;
+        private readonly List<MailRewards> _mailRewardsList = new();
+        private readonly ReactiveProperty<bool> _isDone = new();
+        private readonly List<IDisposable> _disposables = new();
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs:50]: It never dispose. it's ok?

File: nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                var min = i * MaxItemCount;
+                var max = math.min(sortedRewards.Count, (i + 1) * MaxItemCount);
+                for (var j = min; j < max; j++)
+                {
+                    rewards.Add(sortedRewards[j]);
+                }
+
+                item.Set(rewards, textAnchor, offset);
+            }
+
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs:108]:
```suggestion
```

File: nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+            var sortedRewards = mailRewards
+                .OrderBy(x => x.ItemBase.ItemType == ItemType.Consumable)
+                .ThenBy(x => x.ItemBase.ItemType == ItemType.Equipment)
+                .ThenBy(x => x.ItemBase.ItemType == ItemType.Material)
+                .ThenBy(x => x.IsPurchased)
+                .ToList();
+
+            _count = (sortedRewards.Count - 1) / MaxItemCount;
+            var textAnchor = _count == 0 ? TextAnchor.MiddleCenter : TextAnchor.MiddleCenter;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs:97]:
```cs
            var textAnchor = _count == 0 ? TextAnchor.MiddleCenter : TextAnchor.MiddleCenter;
```
It seems `textAnchor` always has `TextAnchor.MiddleCenter`.

File: nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs (hunk 1 above)
Comment (hskim881028) [reply nekoyume/Assets/_Scripts/UI/Widget/Screen/MailRewardScreen.cs:50]: Fixed it!

**Review by sonohoshi — APPROVED** [review]

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**